### DISCUSSION TITLE
How do we change position of robot in examples?

### DISCUSSION THREAD

**Discussion by prajapatisarvesh** [NVlabs/curobo#155] [Software Q&A] [answered]
Hello,

When executing `motion_gen_reacher.py` when I change the position of the robot, it changes the position but the position of collision spheres remains at (0,0,0). And the target works relative to the collision sphere. I couldn't find much in documentation, but has anyone faced similar issue, or have solution for this problem?

Thanks

**Comment by balakumar-s** [comment, marked as answer]
cuRobo plans everything with respect to the robot's base frame. It doesn't have a world frame concept. The collision sphere visualization does not read the current robot pose in the world as it's only for debugging. 

In the examples, obstacles are automatically transformed to the robot's base frame even when you change the robot position. However, the target pose is not transformed. If you can read the robot's base frame and convert the target pose to be in that reference frame, then everything would work.
